sql/core/src/main/scala/org/apache/spark/sql/sources/rules.scala

```diff
@@ -119,6 +120,10 @@ private[sql] case class PreWriteCheck(catalog: Catalog) extends (LogicalPlan =>
         // The relation in l is not an InsertableRelation.
         failAnalysis(s"$l does not allow insertion.")
 
+      case logical.InsertIntoTable(t, _, _, _, _) =>
```

Comment:
Actually, we also use `InsertIntoTable` for `MetastoreRelation`. But, we do not have access to it at here. I think we can match those `LogicalPlan`s that are not `LeafNodes` and also match those `LeafNode`s that do not support insert (e.g. `LocalRelation` and `OneRowRelation`).